org.osgi.test.junit5/src/main/java/org/osgi/test/common/annotation/Property.java

```diff
@@ -55,6 +55,26 @@ public enum Type {
 		PrimitiveArray
 	}
 
+	public enum ValueSource {
+		/** Use the {@link Property#value()} directly */
+		Value,
+		/**
+		 * Use the first {@link Property#value()} as a key in the System
+		 * Properties
+		 */
+		SystemProperty,
+		/**
+		 * Use the first {@link Property#value()} as a key in the Environment
+		 */
+		EnvironmentVariable,
+		/** Use the name of the test class */
+		TestClass,
+		/** Use the name of the test method */
+		TestMethod,
+		/** Use the test unique identifier */
+		TestUniqueId
```

Comment:
Is the unique id for a test fixed so that it can be known _a priori_? The test for this had to go through some contortions to "inject" the uniqueid into the test class. That is, is test unique id a practical source here that people would use? I am not sure.

Reply:
> Is the unique id for a test fixed so that it can be known a priori?

I believe that the answer to this is yes, but it's non-trivial to work out what it will be.

> That is, is test unique id a practical source here that people would use? I am not sure.

It was explicitly requested by @stbischof, but I do wonder if he actually wanted the Display Name. I'll wait for him to weigh in.

Reply:
The case for the TestUniqueId is `running parallel tests`.  I need a discriminator that i can use as ServiceProperty to mark a Service and use the same Id in a reference.target filter to bind the this service to an other.

When Running Parameterized in parallel they will have same TestClass and TestMethod.

Reply:
> The case for the TestUniqueId is running parallel tests. I need a discriminator that i can use as ServiceProperty to mark a Service and use the same Id in a reference.target filter to bind the this service to an other.

This says to me that we should leave the `TestUniqueId` in this PR. I think that the placeholder replacement should be raised as a separate issue so that it doesn't make this PR too big.